services/core/MasterDriverAgent/master_driver/interfaces/modbus_tk/__init__.py

```diff
@@ -0,0 +1,338 @@
+        """
+            Convert original modbus csv format to the new modbus_tk registry_config_lst
+
+        :param old_registry_config_lst: list of all register dictionaries in old volttron csv format
+
+        :type old_registry_config_lst: list
+        """
+        new_registry_config_lst = []
+        for reg_dict in old_registry_config_lst:
+            point_name = reg_dict.get('Volttron Point Name')
+            register_name = reg_dict.get('Reference Point Name', point_name).replace(" ", "_").lower()
+            address = reg_dict.get('Point Address')
+
+            csv_type = reg_dict['Modbus Register']
+            datatype = type_map.get(csv_type[1:], None)
+            if datatype is None: # for string[len] or BOOL type
+                datatype = csv_type.lower()
+
+            unit = reg_dict.get('Units')
+            writable = reg_dict.get('Writable')
+            default_value = reg_dict.get('Default Value', None)
+            description = reg_dict.get('Note', '')
+
+            new_registry_config_lst.append({'Volttron Point Name': point_name,
+                                            'Register Name': register_name,
+                                            'Address': address,
+                                            'Type': datatype,
+                                            'Units': unit,
+                                            'Writable': writable,
+                                            'Default Value': default_value,
+                                            'Description': description})
+
+        return new_registry_config_lst
+
+    def configure(self, config_dict, registry_config_lst):
+        """
+            Parse driver and csv config to define client transport, add registers to ModbusTKRegister,
+            and set default values for revert reading
+
+        :param config_dict: dictionary of device configure
+        :param registry_config_lst: the list of all register dictionary parsed from the csv file
+
+        :type config_dict: dictionary
+        :type registry_config_lst: list
+        """
+        name = config_dict.get('name', 'UNKOWN')
+        device_address = config_dict['device_address']
+        port = config_dict.get('port', None)
+        slave_address = config_dict.get('slave_id', 1)
+        baudrate = config_dict.get('baudrate', 9600)
+        bytesize = config_dict.get('bytesize', 8)
+        parity = parity_map[config_dict.get('parity', 'none')]
+        stopbits = config_dict.get('stopbits', 1)
+        xonxoff = config_dict.get('xonxoff', 0)
+        addressing = config_dict.get('addressing', OFFSET).lower()
+        endian = endian_map[config_dict.get('endian', BIG)]
+        write_single_values = not str2bool(str(config_dict.get('write_multiple_registers', "True")))
+
+        # Convert original modbus csv config format to the new modbus_tk registry_config_lst
+        if registry_config_lst and 'Point Address' in registry_config_lst[0].keys():
```

Comment:
It is not necessary to call keys() to create a list as "in" will look for the value in the dictionary keys already.

Reply:
Thanks. I fixed it.